Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/SiteResolver.kt

```diff
@@ -27,20 +27,17 @@ open class SiteResolver @Inject constructor(
 ) {
 
   open fun findSiteForUrl(url: String): Site? {
-    var httpUrl = sanitizeUrl(url)
-
-    if (httpUrl == null) {
-      return siteManager.firstActiveSiteOrNull { _, site ->
+    var httpUrl: HttpUrl? = sanitizeUrl(url)
+      ?: return siteManager.firstActiveSiteOrNull { _, site ->
```

Comment:
Uhh, I don't really like using "?: return" construction when there is a lot of code after the "return" statement (like in this case) because, in my opinion, it's way harder to read. That's why I disabled this inspection locally. But I think it's find to use it to return some value/throw an exception, basically do any operation that takes 1 line (well maybe 2 in some cases). 

Like here it's fine to use it:
```
val boards = boardsMap[siteDescriptor]?.values
          ?: return
```

but here it makes it harder to read:
```
val boards = boardsMap[siteDescriptor]?.values
          ?: return someOtherValue?.let {
    <...> lots of code here <...>
} 
```

so I would use the good old if check
```
val boards = boardsMap[siteDescriptor]?.values
if (boards == null) {
    return someOtherValue?.let {
        <...> lots of code here <...>
    } 
}
```

I think it's better to just disable this inspection globally for the whole project and just forget about it.

Reply:
You can also remove the ": HttpUrl?" and get rid of "!!" operator.

Reply:
Alright, will do.